### ISSUE TITLE
Bug Report: using 1.75-1.77 docker is not available, but it doesn't seem to be deprecated at the moment

### ISSUE THREAD

**Issue by alekspickle** [CircleCI-Public/cimg-rust#137]
*Note: We also welcome PRs to fix bugs! This helps us take action faster where a bug has been identified!*

For our official CircleCI Docker Convenience Image support policy, please see [CircleCI docs](https://circleci.com/docs/convenience-images-support-policy).

This policy outlines the release, update, and deprecation policy for CircleCI Docker Convenience Images.

---

**Describe the bug**
I am using cimg-rust:1.75.0 and in one of the long running workflows without any changes in circleci config I am suddenly getting:
`docker: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?`

**To Reproduce**
Try to spin up docker container with `cimg/rust:1.75.0`.

**Expected behavior**
Docker container is up and running for a job

**Workarounds**
not really

**Screenshots and Build Links**
I don't think it will help the error is pretty clear - there are no docker available, however there should be because on the list of tools of image 1.75 docker should be 24.0.6

**Additionat context**
If I re-run with ssh and try to run docker stats I get the same:
```
circleci@d97249c19321:~$ docker stats --no-stream
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
circleci@d97249c19321:~$ ls /var/run/
lock  log  mount  sendsigs.omit.d  shm  sudo  systemd  user
circleci@d97249c19321:~$ service docker status
 * Docker is not running
```
So basically - docker cli is present, but docker socket is not created(docker did not start)

**Comment by alekspickle** [comment]
nevermind, it was in fact setup_remote_docker. :facepalm:
it was just not clear because of how complicated our cci config is
